.github/workflows/update-buildconfigs.yaml

```diff
@@ -0,0 +1,183 @@
+# The aim of this GitHub workflow is to update the cuda-rstudio-buildconfig.yaml and rstudio-buildconfig.yaml
+# If No version is provided, it will update the BuildConfigs to the next version by incrementing the minor version number by 1.
+---
+name: Update BuildConfigs
+on:
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: "Optional: Provide branch name"
+        required: false
+        type: string
+      version:
+        description: "Optional: Provide version to update BuildConfigs to"
+        required: false
+        type: string
+  workflow_call:
+    inputs:
+      branch:
+        required: false
+        type: string
+      version:
+        required: false
+        type: string
+env:
+  TMP_BRANCH: tmp-verify-${{ github.run_id }}
+  BRANCH: ${{ inputs.branch || 'main' }}
+  VERSION: ${{ inputs.version }}
+
+jobs:
+  buidConfigs:
+    runs-on: ubuntu-latest
+    if: inputs.version == ''
```

Comment:
I wonder whether instead of having different job for each variant meaning having the duplicate code, we couldn't have an extra step that would define a new variable depending on the `inputs.version` value - either use it directly or count a new value for us. This new variable would then be used by the followup steps. WDYT?